### PR TITLE
Fix a null pointer exception

### PR DESCRIPTION
**Pull request by bzheng888**
### What changes are proposed in this pull request?
Fix a null pointer problem in journal system.
### Why are the changes needed?
Please clarify why the changes are needed. For instance,
When construct the RaftJournalSystem, the member mStateMachine is null pointer, because the registerMetrics is called before mStateMachine initialized。
### Does this PR introduce any user facing changes?
No
Please list the user-facing changes introduced by your change, including
No

### CODE REVIEW

**Review by apc999 — APPROVED** [review]

File: core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java [hunk 1]
```diff
@@ -527,8 +527,11 @@ public synchronized void losePrimacy() {
 
   @Override
   public synchronized Map<String, Long> getCurrentSequenceNumbers() {
-    long currentGlobalState = mStateMachine.getLastAppliedSequenceNumber();
     Map<String, Long> sequenceMap = new HashMap<>();
+    if (mStateMachine == null) {
```
Comment (ggezer) [inline core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:531]: We shouldn't return empty sequences. Please replace this with below:

`Preconditions.checkState(mStateMachine != 1, "State machine not initialized");
`

**Review by ggezer — CHANGES_REQUESTED** [review]

File: core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java (hunk 1 above)
Comment (apc999) [reply core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:531]: then is this more like a bug and check why `mStateMachine` is `null`?
@bzheng888  have you seen this in your environment that `mStateMachine` is `null`?

Comment (bzheng888) [reply core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:531]: > then is this more like a bug and check why `mStateMachine` is `null`?
> @bzheng888 have you seen this in your environment that `mStateMachine` is `null`?
Yeah, I debugged in my environment and it's really a must present bug, I think the registerMetrics() in the construct method  is useless, the metrics is never taken effect.

Comment (bzheng888) [reply core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:531]: Yeah, I debugged in my environment and it's really a must present bug, I think the registerMetrics() in the construct method is useless, the metrics is never taken effect.

Comment (maobaolong) [reply core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:531]: @bzheng888 So would you like to move the `registerMetrics` to the end of `initServer` ?

**Comment by bzheng888** [comment]
@apc999 @ggezer @maobaolong Move the call of registerMetrics() after stateMachine initialized,  this fix is more appropriate, PTAL!

**Review by ggezer — APPROVED** [review]

File: core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java [hunk 2]
```diff
@@ -401,6 +396,7 @@ private synchronized void initServer() throws IOException {
         .setProperties(properties)
         .setParameters(parameters)
         .build();
+    super.registerMetrics();
```
Comment (ggezer) [inline core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:399]: is it OK to reregister at every gainPrimacy() call?

**Review by ggezer — CHANGES_REQUESTED** [review]

File: core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java (hunk 2 above)
Comment (bzheng888) [reply core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:399]: image
the register strategy in registerMetrics() is ifAbsent, so it doesn't seem to reregister every time.

**Comment by apc999** [comment]
any more comments, @ggezer ?

File: core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java (hunk 2 above)
Comment (ggezer) [reply core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java:399]: OK. Still, please move this line after `initServer();` call in `RaftJournalSystem.startInternal()` method.

**Comment by bzheng888** [comment]
@ggezer Done, PTAL!

**Review by ggezer — APPROVED** [review]
Thanks for the fix.

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Pull request merged**